upgrade-instructions.md

```diff
@@ -25,7 +25,6 @@ Change list:
  - [JAMES-2703 Post 3.4.0 release removals](#james-2703-post-340-release-removals)
  - [Health checks routes return code changes](#health-checks-routes-return-code-changes)
  - [User mailboxes reIndexing endpoint change](#user-mailboxes-reindexing-endpoint-change)
- - [Hybrid blobStore replaces Union blobStore](#hybrid-blobstore-replaces-union-blobstore)
```

Comment:
I don't think this should be deleted... But add a new entry to say we remove it for next version. 

I think though we forgot to modify this when we released 3.5? The unreleased section is still aboutchanges between james 3.4 and 3.5

Reply:
> I think though we forgot to modify this when we released 3.5? The unreleased section is still aboutchanges between james 3.4 and 3.5

Release is not yet official.